app/src/lib/stores/app-store.ts

```diff
@@ -2519,27 +2521,46 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.updateMenuItemLabels(latestState)
 
     this._initializeCompare(repository)
-    this.refreshIndicatorsForRepositories([repository], false)
+    this.updateSidebarIndicator(repository, status)
   }
 
-  public refreshAllIndicators() {
-    return this.refreshIndicatorsForRepositories(this.repositories, true)
+  /**
+   * Update the repository sidebar indicator for the repository
+   */
+  private async updateSidebarIndicator(
+    repository: Repository,
+    status: IStatusResult | null
+  ): Promise<void> {
+    const lookup = this.localRepositoryStateLookup
+
+    if (repository.missing) {
+      lookup.delete(repository.id)
+      return
+    }
+
+    if (status === null) {
+      lookup.delete(repository.id)
+      return
+    }
+
+    lookup.set(repository.id, {
+      aheadBehind: status.branchAheadBehind || null,
+      changedFilesCount: status.workingDirectory.files.length,
+    })
   }
 
   /**
-   * Refresh in-memory indicators for a set of repositories
-   *
-   * @param repositories the set of repositories to update
-   * @param tryBackgroundFetch whether the action should also try and fetch new changes from the remote
+   * Refresh sidebar indicators for the set of repositories tracked in the app.
    */
-  private async refreshIndicatorsForRepositories(
-    repositories: ReadonlyArray<Repository>,
-    tryBackgroundFetch: boolean
-  ): Promise<void> {
+  public async refreshAllSidebarIndicators() {
     const startTime = performance && performance.now ? performance.now() : null
 
+    // keep a reference to the current set of repositories to avoid the array
+    // changing while this is running
+    const repositories = new Array<Repository>(...this.repositories)
+
     for (const repo of repositories) {
-      await this.refreshIndicatorForRepository(repo, tryBackgroundFetch)
+      await this.refreshIndicatorForRepository(repo)
```

Comment:
that _might_ be risky if there are a large number of repositories (since we'd flood the process with promises), so i think i have a slight preference for the original implementation.

Reply:
We talked about this in #5312 as there were some performance issue when we initially shipped the repository indicators - we'd create a number promises in the background to do work, and for users with dozens of repositories it would significantly affect things with no indicator of why.

A `for-loop` is fine here.